### PR TITLE
Add special x86-SSE2-only build

### PR DESCRIPTION
**Pull request by rschu1ze**
See https://github.com/ClickHouse/ClickHouse/issues/40459#issuecomment-1236464315

### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Extended the CI configuration to create a x86 SSE2-only build. Useful for old or embedded hardware.

### CODE REVIEW

**Comment by nickitat** [comment]
the main question is probably how do we guarantee that this binary won't fail with illegal instruction. maybe we just will not provide this guarantee.

File: src/Functions/GatherUtils/sliceHasImplAnyAll.h [hunk 1]
```diff
@@ -419,7 +419,7 @@ DECLARE_SSE42_SPECIFIC_CODE (
 // SSE4.2 Int64, UInt64 specialization
 template<typename IntType>
 requires (std::is_same_v<IntType, Int64> || std::is_same_v<IntType, UInt64>)
-inline ALWAYS_INLINE bool sliceHasImplAnyAllImplInt64(
+inline bool sliceHasImplAnyAllImplInt64(
```
Comment (nickitat) [inline src/Functions/GatherUtils/sliceHasImplAnyAll.h:422]: why these changes are needed?

Comment (rschu1ze) [reply src/Functions/GatherUtils/sliceHasImplAnyAll.h:422]: Explained in commit msg header [here](https://github.com/ClickHouse/ClickHouse/pull/40999/commits/d054ffd1109ab93ec366a64558c7c983cd23a11e).

**Comment by rschu1ze** [comment]
> the main question is probably how do we guarantee that this binary won't fail with illegal instruction. maybe we just will not provide this guarantee.

Hmmm, I wonder how it could fail with illegal instruction? That would only happen if someone runs it on an antique CPU w/o SSE2.

File: src/Functions/GatherUtils/sliceHasImplAnyAll.h (hunk 1 above)
Comment (Algunenano) [reply src/Functions/GatherUtils/sliceHasImplAnyAll.h:422]: Isn't this code only used under `USE_MULTITARGET_CODE`? I think it'd make more sense to keep the inlines and disable `ENABLE_MULTITARGET_CODE` for this kind of build, right?

Comment (rschu1ze) [reply src/Functions/GatherUtils/sliceHasImplAnyAll.h:422]: Yeah, that's the other option I thought about.  My feeling is that removing `FORCE_INLINE` is the cleaner approach. It avoids leaking a instruction set requirement from the optimized code into the generic code. But I also see that quite a few queries slowed down 😞 
Okay, will disable `ENABLE_MULTITARGET_CODE` then ...

**Comment by nickitat** [comment]
> Hmmm, I wonder how it could fail with illegal instruction? That would only happen if someone runs it on an antique CPU w/o SSE2.

my bad, I had an impression that compilers enable more than sse2 by default.

**Review by nickitat — APPROVED** [review]
lgtm

**Review by alexey-milovidov — APPROVED** [review]
